### PR TITLE
DB-23237 Superset: datatypes are not being converted properly

### PR DESCRIPTION
**Pull request by alexclavel-ocient**
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/ocient.py [hunk 1]
```diff
@@ -237,10 +237,12 @@ def fetch_data(cls, cursor, lim=None):
...
-                    row[info.column_index] = info.sanitize_func(v)
-
+            def do_nothing(x): 
+                return x
+            
+            sanitization_functions = [do_nothing for _ in range(len(rows[0]))]
+            for info in columns_to_sanitize:
+                sanitization_functions[info.column_index] = info.sanitize_func
+            
+            rows = [[sanitization_functions[i](row[i]) for i in range(len(row))] for row in rows]
```
Comment (jwilliams-ocient) [inline superset/db_engine_specs/ocient.py:248]: (nit): add a comment specifying why we can't update in place. Secondly, is there a reason to keep the records immutable? i.e PEP249 spec

**Review by jwilliams-ocient — APPROVED** [review]

File: superset/db_engine_specs/ocient.py [hunk 2]
```diff
@@ -237,10 +237,12 @@ def fetch_data(cls, cursor, lim=None):
...
-            for row in rows:
-                for info in columns_to_sanitize:
-                    # Modify the element in-place.
-                    v = row[info.column_index]
-                    row[info.column_index] = info.sanitize_func(v)
-
+            def do_nothing(x): 
+                return x
+            
+            sanitization_functions = [do_nothing for _ in range(len(rows[0]))]
```
Comment (jwilliams-ocient) [inline superset/db_engine_specs/ocient.py:243]: is the row list also immutable?

Comment (jwilliams-ocient) [inline superset/db_engine_specs/ocient.py]: is the row list also immutable?

File: superset/db_engine_specs/ocient.py (hunk 1 above)
Comment (alexclavel-ocient) [reply superset/db_engine_specs/ocient.py:248]: No reason to make immutable, this is just being passed to a json parser on superset's side.

File: superset/db_engine_specs/ocient.py (hunk 2 above)
Comment (jwilliams-ocient) [reply superset/db_engine_specs/ocient.py]:
```python
funcs = to_dict(columns_to_sanitize)

for row_idx, row in enumerate(rows):
  rows[row_idx] = [
    funcs[col_idx].sanitize_func(val) 
    for col_idx, val in enumerate(row) 
    if col_idx in funcs 
    else val
  ]
```
Another alternative, though I'm not sure it's any more readable...

Comment (alexclavel-ocient) [reply superset/db_engine_specs/ocient.py]: No it is not. It is simply a list

**Review by jwilliams-ocient — APPROVED** [review]

**Pull request merged**